source/Nevermore/IWriteTransaction.cs

```diff
@@ -5,6 +5,8 @@ namespace Nevermore
 {
     public interface IWriteTransaction : IReadTransaction, IWriteQueryExecutor
     {
+        void CommitIfOwned();
+        Task CommitIfOwnedAsync(CancellationToken cancellationToken = default);
```

Comment:
Per the feedback on https://github.com/OctopusDeploy/OctopusDeploy/pull/23120, we'd rather not introduce a different way to commit a Nevermore transaction. Rather, the consumer (Octopus Server) should be aware of whether the SQL transaction is externally owned and simply not call `Commit()` on the Nevermore transaction in that case. Happy to assist with ways to solve that.